plugins/plugin-installer/plugin_installer/tasks.py

```diff
@@ -110,57 +126,85 @@ def extract_module_paths(url):
     return module_paths
 
 
-def extract_plugin_name(plugin_url):
+def extract_plugin_dir(plugin_url):
+    try:
+        # download and unpack plugin_url
+        plugin_dir = tempfile.mkdtemp()
+        req_set = pip.req.RequirementSet(build_dir=None,
+                                         src_dir=None,
+                                         download_dir=None)
+        req_set.unpack_url(link=pip.index.Link(plugin_url),
+                           location=plugin_dir,
+                           download_dir=None,
+                           only_download=False)
+        return plugin_dir
+    except Exception as e:
+        if os.path.exists(plugin_dir):
```

Comment:
'plugin_dir' may not be defined if tempfile.mkdtemp() fails. its a very unlikely scenario, but still its better to be protected from it.